data/blockchain/blockchain.go

```diff
@@ -37,6 +40,7 @@ func NewBlockChain(
 		localHeight:        -1,
 		networkHeight:      -1,
 		badBlocks:          badBlocksCache,
+		appStatusHandler:   appStatusHandler,
```

Comment:
the check from line 101 can be moved in this constructor func

Reply:
done